LibraryAPI.java

```diff
@@ -0,0 +1,187 @@
+        Author authors[] = new Author[10];
+        for (int i = 0; i < authors.length; i++) {
+            Author author = new Author(LibraryDataRepo.names[(int) (Math.random() * 7)], LibraryDataRepo.surnames[(int) (Math.random() * 7)], (int) (Math.random() * 100));
+            authors[i] = author;
+        }
+        return authors;
+    }
+
+    public static Book[] getBookData(Author authors[]) {
+        Book books[] = new Book[10];
+        for (int i = 0; i < books.length; i++) {
+            Book book = new Book(LibraryDataRepo.titles[(int) (Math.random() * 5)],
+                    authors[(int) (Math.random() * authors.length)],
+                    Book.createRandomDate(1900, 2000),
+                    LibraryDataRepo.categories[(int) (Math.random() * 5)],
+                    (float) Math.random() * 10);
+            books[i] = book;
+        }
+        return books;
+    }
+
+    public static int getGivenBooksLength(Book[] books, Author author) {
+        int length = 0;
+        for (int i = 0; i < books.length; i++) {
+            if (author.equals(books[i].getAuthor())) {
+                length++;
+            }
+        }
+        return length;
+    }
+
+    public static Book[] getGivenAuthorBooks(Book books[], Author author) {
+        Book givenBooks[] = new Book[getGivenBooksLength(books, author)];
+        int j = 0;
+        for (int i = 0; i < books.length; i++) {
+            if (author.equals(books[i].getAuthor())) {
+                givenBooks[j++] = books[i];
+            }
+        }
+        return givenBooks;
+    }
+
+    public static boolean isExistNameAndSurnameInRepo(String name, String surname) {
+        for (int i = 0; i < LibraryDataRepo.names.length; i++) {
+            if (name.equals(LibraryDataRepo.names[i])) {
+                for (int j = 0; j < LibraryDataRepo.surnames.length; j++) {
+                    if (surname.equals(LibraryDataRepo.surnames[i])) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    public static Book[] getAllBooksOfOneAuthor(String name, String surname) {
+        Book allBooks[] = new Book[10];
+        if (isExistNameAndSurnameInRepo(name, surname)) {
+            for (int i = 0; i < allBooks.length; i++) {
+                Book book = new Book(LibraryDataRepo.titles[(int) (Math.random() * 5)],
+                        new Author(name, surname, 78),
+                        Book.createRandomDate(1900, 2000),
+                        LibraryDataRepo.categories[(int) (Math.random() * 5)],
+                        (float) Math.random() * 10);
+                allBooks[i] = book;
+            }
+        }
+        return allBooks;
+    }
+
+    public static Book[] getTopThreeBooks(Book allBooks[]) {
+        Book topThreeBooks[] = new Book[3];
+        int j = 0;
+        Arrays.sort(allBooks, LibraryAPI::compareBooks);
+        for (int i = 6; i >= 4; i--) {

```

Comment:
don't use literal numbers, use allbooks.length as base of your calculations instead